### DISCUSSION TITLE
Plans to migrate P tools to .NET 6 SDK and beyond?

### DISCUSSION THREAD

**Discussion by dasblinkenlight** [p-org/P#473] [Q&A] [answered]
Currently, P uses the specific version of .Net SDK 3.1, which is nearing its end-of-support date of Dec-13, 2022. Migrating to the current version of .NET, i.e. the .NET 6, would be desirable, because it's the first version of .NET that supports Apple's M1 natively.

Do you have plans to migrate to .NET 6 or its successors? What is involved in migrating to .NET 6? Do you need help updating or testing the code?

**Comment by ankushdesai** [comment, marked as answer]
Hi @dasblinkenlight, thanks for the question. Yes, we are currently working on upgrading P to .Net 6. You can expect this new version 1.2 of P to be released by the end of July.

It would be great help if you can test and validate the new version after its released. Any help from the community would be great! 

Thanks again!

**Reply by dasblinkenlight** [reply]
Thank you very much! I opened a [pull request 475](https://github.com/p-org/P/pull/475) which updates the project files and the templates of project files used for generating unit tests. Most changes are mechanical except the work-around to a Visual Studio problem with the P compiler project, where ANTLR-generated code was default-included twice (see the DefaultExcludesInProjectFolder entry in the project file).
